martypy/WebSocket.py

```diff
@@ -67,7 +77,36 @@ def _sendUpgradeReq(self) -> None:
 
     def service(self) -> None:
         # Get any data
-        rxData = self.sock.recv(self.maxSocketBytes)
+        reconnectRequired = False
+        try:
+            rxData = self.sock.recv(self.maxSocketBytes)
+        except OSError as excp:
+            logger.debug(f"WebSocket OSError {excp}")
+            if not self.autoReconnect:
+                raise excp
+            reconnectRequired = True
+        except Exception as excp:
+            logger.debug(f"WebSocket exception {excp}")
+            if not self.autoReconnect:
+                raise excp
+            reconnectRequired = True
+        # Check for reconnect
+        if reconnectRequired:
+            if self.reconnectLastTime is None or time.time() > self.reconnectLastTime + self.reconnectRepeatSecs:
+                self.close()
+                try:
+                    if self.onReconnect:
+                        self.onReconnect()
+                    self.open()
+                    logger.debug("WebSocket reopened automatically")
+                except Exception as excp:
+                    logger.debug("WebSocket exception trying to reopen websocket", exc_info=True)
+                self.reconnectLastTime = time.time()
+            else:
+                time.sleep(0.01)
+            return
```

Comment:
I think this could be its own function, no? We could then just call it when appropriate instead of using the `reconnectRequired` variable which would save a level of nesting and make the code more readable.

Reply:
Sure, if you want to make that change and check it works then I'm happy.